plugins/EventGhost/__init__.py

```diff
@@ -608,20 +644,75 @@ class text:
         text1 = "Event string to fire:"
         text2 = "Delay the firing of the event:"
         text3 = "seconds. (0 = fire immediately)"
+        text4 = "Add event to event queue:"
+        text5 = "Return eg.event to original event:"
 
-    def __call__(self, eventString, waitTime=0):
+    def __call__(
+        self,
+        eventString,
+        waitTime=0,
+        queueEvent=True,
+        restoreEvent=False
+    ):
         eventString = eg.ParseString(eventString)
         if not waitTime:
-            eg.TriggerEvent(eventString)
+            if queueEvent:
+                eg.TriggerEvent(eventString)
+            else:
+                event = eg.EventGhostEvent(suffix=eventString)
+                if restoreEvent:
+                    old_event_string = eg.eventString
+                    old_event = eg.event
+                    event.Execute()
+                    eg.event = old_event
+                    eg.eventString = old_event_string
+                else:
+                    event.Execute()
         else:
             eg.scheduler.AddShortTask(waitTime, eg.TriggerEvent, eventString)
 
-    def Configure(self, eventString="", waitTime=0):
+    def Configure(
+        self,
+        eventString="",
+        waitTime=0,
+        queueEvent=False,
+        restoreEvent=False
+    ):
         panel = eg.ConfigPanel()
         text = self.text
 
         eventStringCtrl = panel.TextCtrl(eventString, size=(250, -1))
         waitTimeCtrl = panel.SpinNumCtrl(waitTime, integerWidth=5)
+        queueEventCtrl = wx.CheckBox(panel, -1, '')
+        restoreEventCtrl = wx.CheckBox(panel, -1, '')
+
+        queueEventCtrl.SetValue(queueEvent)
+        restoreEventCtrl.SetValue(restoreEvent)
+        queueEventText = panel.StaticText(text.text4)
+        restoreEventText = panel.StaticText(text.text5)
+
+        def on_spin(evt):
+            def check_spin():
+                value = bool(waitTimeCtrl.GetValue())
+                queueEventText.Show(not value)
+                queueEventCtrl.Show(not value)
+                restoreEventText.Show(
+                    not value or (not value and not queueEventCtrl.GetValue())
+                )
+                restoreEventCtrl.Show(
+                    not value or (not value and not queueEventCtrl.GetValue())
+                )
+            wx.CallLater(20, check_spin).Start()
```

Comment:
What does the ``.Start()` do?

Reply:
I have gotten into the habit of using call later whenever i am checking a wx widget for a value that can be changed by keyboard input as an example if you have a TextCtrl that you want to make sure it is only digits that are entered and you want to not allow the input if it's not. I know this can be done on a per character basis which i feel is more complex trying to use keycodes and what have you to determine this. so if you bind to EVT_CHAR or EVT_CHARHOOK if you grab the value of the control to be able to parse it it will not reflect the current keyboard input that triggered the event. but if you use a CallLater which is basically the same thing as CallAfter except it has a timer for when it should run you need to use Start() to start the timer. but you can call a function that would check and parse the input as needed and it would now be updated so when you call GetValue() on the control it will have been updated with the latest keystrokes.

Reply:
You don't need to add Start() to CallLater(). In CallLater's __init__ method Start() will automatically be called.

Reply:
really?!?! this is did not know.